File: app/src/main/java/com/sevtinge/cemiuiler/module/hook/securitycenter/other/DisableRootCheck.java
package com.sevtinge.cemiuiler.module.hook.securitycenter.other;

import com.sevtinge.cemiuiler.module.base.BaseHook;
import com.sevtinge.cemiuiler.module.hook.securitycenter.SecurityCenterDexKit;

import java.lang.reflect.Method;
import java.util.List;
import java.util.Objects;

import de.robv.android.xposed.XC_MethodReplacement;
import de.robv.android.xposed.XposedBridge;
import io.luckypray.dexkit.descriptor.member.DexMethodDescriptor;

public class DisableRootCheck extends BaseHook {
    @Override
    public void init() {
        try {
            List<DexMethodDescriptor> result = Objects.requireNonNull(SecurityCenterDexKit.mSecurityCenterResultMap.get("rootCheck"));
            for (DexMethodDescriptor descriptor : result) {
                Method checkIsRooted = descriptor.getMethodInstance(lpparam.classLoader);
                if (checkIsRooted.getReturnType() == boolean.class) {
                    XposedBridge.hookMethod(checkIsRooted, XC_MethodReplacement.returnConstant(false));
                }
            }
        } catch (Throwable e) {
            e.printStackTrace();
        }
    }


}

File: app/src/main/java/com/sevtinge/hyperceiler/module/hook/securitycenter/other/DisableRootCheck.kt
package com.sevtinge.hyperceiler.module.hook.securitycenter.other

import com.github.kyuubiran.ezxhelper.HookFactory.`-Static`.createHook
import com.sevtinge.hyperceiler.module.base.BaseHook
import com.sevtinge.hyperceiler.utils.DexKit.addUsingStringsEquals
import com.sevtinge.hyperceiler.utils.DexKit.dexKitBridge

object DisableRootCheck : BaseHook() {
    override fun init() {
        dexKitBridge.findMethod {
            matcher {
                addUsingStringsEquals("key_check_item_root")
                returnType = "boolean"
            }
        }.single().getMethodInstance(lpparam.classLoader)?.createHook {
            returnConstant(false)
        }
    }
}
